Make image build and rrun cancelable

Xarthisius opened the pull request:
This adds checks for `task.canceled` in long running tasks: `image_build` and `recorded_run`. In the latter case also tries to clean-up leftovers nicely.

### How to test?
1. Create a Tale, build image (preferably with a custom Dockerfile so you don't hit cache).
2. In girder UI navigate to job view (click on job name in notification). Click 'cancel'. Job should turn to "Cancelling" than "Canceled" after a while.
3. Create a Tale, upload `entrypoint.sh` with steps that take some time (e.g. `echo "1" && sleep 10; echo "2" && sleep 10;...`). Run recorded run. Repeat 2. In dashboard UI run should also be in "Canceled" state in the end.

### TODO:
- [ ] tests...

craig-willis commented:
Two observations:
* When I cancel an image build, the Build Tale Image job cancels, but the chained jobs still proceed:
![Screen Shot 2022-12-14 at 10 00 34 AM](https://user-images.githubusercontent.com/4334350/207630684-22454823-4617-4e5a-958d-71a3b18f397a.png)
* After canceling an RR job, I noticed that the wt_progress event state returned to the UI is `error`. Looking at the[ jobs plugin](https://github.com/whole-tale/girder/blob/master/plugins/jobs/server/constants.py#L60), they don't seem to offer a `canceled` state?  I was thinking it would be nice to display a canceled message on the notification panel instead of error.

Xarthisius commented:
> * When I cancel an image build, the Build Tale Image job cancels, but the chained jobs still proceed:
>   ![Screen Shot 2022-12-14 at 10 00 34 AM](https://user-images.githubusercontent.com/4334350/207630684-22454823-4617-4e5a-958d-71a3b18f397a.png)

Fixed by https://github.com/whole-tale/girder_wholetale/pull/565

> * After canceling an RR job, I noticed that the wt_progress event state returned to the UI is `error`. Looking at the[ jobs plugin](https://github.com/whole-tale/girder/blob/master/plugins/jobs/server/constants.py#L60), they don't seem to offer a `canceled` state?  I was thinking it would be nice to display a canceled message on the notification panel instead of error.

Fixed by https://github.com/whole-tale/ngx-dashboard/pull/310/commits/e7d67c710ac04d68d81c13de84df7b6a03652cc7

craig-willis reviewed (approved):
wfm
